tests/mamba/test_checkpointing_ssu.py

```
@@ -4593,3 +4593,176 @@ def test_checkpointing_ssu_noncontig(state_dtype, varlen):
+    cache_size = batch
+    assert (prev_k + npredicted > max_window) == must_checkpoint, (
+        f"test bug: parametrize disagrees with must_checkpoint formula "
+        f"(prev_k={prev_k}, npredicted={npredicted}, max_window={max_window})"
+    )
+
+    def _hash(t):
+        t = t.detach().cpu().contiguous()
+        # numpy can't view bf16/fp8 directly — promote losslessly to f32.
+        if t.dtype in (torch.bfloat16, torch.float8_e4m3fn):
+            t = t.float()
+        return hashlib.sha1(t.numpy().tobytes()).hexdigest()
```

Ruff (0.15.14)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)